flax/linen/module.py

```diff
@@ -798,7 +827,16 @@ def apply(self, variables: VariableDict, *args, rngs: RNGSequences = None,
     else:
       method = _get_unbound_fn(method)
     fn = lambda scope: method(self.clone(parent=scope), *args, **kwargs)
-    return apply(fn, mutable=mutable)(variables, rngs=rngs)
+    if capture_intermediates is True:
+      capture_intermediates = capture_call_intermediates
+    if capture_intermediates:
+      mutable = union_filters(mutable, 'intermediates')
+    _context.capture_stack.append(capture_intermediates)
```

Comment:
Why do we need a capture stack? Can there be multiple "capture intermediate" scopes nested into each other?

Reply:
If you use apply inside apply yes

Reply:
Ah I see. (Though I'm curious how that actually works [at least within nn.compact] because once you instantiate a module it's a child of the current module...)